application_form/services/application.py

```diff
@@ -309,9 +309,8 @@ def _cancel_lower_priority_haso_applications(
 ) -> None:
     """
     Go through the given winning applications, and cancel each application made for
-    an apartment that has a lower priority than the reserved apartment and is not in
-    the first place in the queue. The canceled application is removed from the queue of
-    the corresponding apartment.
+    an apartment that has a lower priority than the reserved apartment.
```

Comment:
This could also mention that it cancels only reservations in state RESERVED, and not ones in "further" states, like OFFERED.